src/components/utils/test/thread_test.cc

```diff
@@ -72,6 +74,76 @@ class TestDelegate : public threads::ThreadDelegate {
   bool check_value_;
 };
 
+class InfiniteMainTestDelegate : public threads::ThreadDelegate {
+ public:
+  InfiniteMainTestDelegate()
+      : notified_(false), notify_running_(false), check_value_(false) {}
+
+  void threadMain() {
+    {
+      AutoLock test_lock(test_mutex_);
+      check_value_ = true;
+      notify_running_ = true;
+      cond_var_.Broadcast();
+    }
+
+    while (!notified_) {
+      threads::sleep(1);
+    }
+  }
+
+  bool check_value() const {
+    return check_value_;
+  }
+
+  void exitThreadMain() {
+    notified_ = true;
+  }
+
+  volatile bool notified_;
+  volatile bool notify_running_;
+
+ private:
+  bool check_value_;
+};
+
+class DelayedExitTestDelegate : public threads::ThreadDelegate {
+ public:
+  DelayedExitTestDelegate()
+      : notified_main_(false), notified_exit_(false), notify_running_(false) {}
+
+  void threadMain() {
+    {
+      AutoLock test_lock(test_mutex_);
+      notify_running_ = true;
+      cond_var_.Broadcast();
+    }
+
+    while (!notified_main_) {
+      threads::sleep(1);
+    }
+
+    {
+      AutoLock test_lock(test_mutex_);
+      notify_running_ = true;
+      cond_var_.Broadcast();
+    }
+    notified_exit_ = true;
+  }
+
+  void exitThreadMain() {
+    notified_main_ = true;
```

Comment:
@pvvasilev So here you can change the flag and notify conditional variable. So your `threadMain` function will be notified,